### PR TITLE
[BUGFIX] raise error if no identifier provided to get checkpoint/expectation suite

### PR DESCRIPTION
**Pull request by DrewHoo**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], or [CONTRIB]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- raise a ValueError if get_expectation_suite or get_checkpoint called for CloudDataContext with no identifiers

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in GitHub issues or Slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [ ] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [ ] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [ ] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/data_context/data_context/abstract_data_context.py [hunk 1]
```diff
@@ -2020,10 +2020,14 @@ def get_checkpoint(
...
 
         Raises:
-            CheckpointNotFoundError: If the requested Checkpoint does not exists.
+            CheckpointNotFoundError: If the requested Checkpoint does not exist.
         """
         # <GX_RENAME>
         id = self._resolve_id_and_ge_cloud_id(id=id, ge_cloud_id=ge_cloud_id)
+
+        if not name and not id:
+            raise ValueError(f"name and id cannot both be None")
```
Comment (cdkini) [inline great_expectations/data_context/data_context/abstract_data_context.py]: Don't need the f-string here

File: tests/data_context/cloud_data_context/test_expectation_suite_crud.py [hunk 2]
```diff
@@ -436,6 +436,19 @@ def test_get_expectation_suite_nonexistent_suite_raises_error(
...
+@pytest.mark.cloud
+def test_get_expectation_suite_no_identifier_raises_error(
+    empty_base_data_context_in_cloud_mode: CloudDataContext,
+) -> None:
+    context = empty_base_data_context_in_cloud_mode
+
+    suite_id = None
+
+    with pytest.raises(ValueError):
+        context.get_expectation_suite(ge_cloud_id=suite_id)
```
Comment (cdkini) [inline tests/data_context/cloud_data_context/test_expectation_suite_crud.py]:
```suggestion
        context.get_expectation_suite()
```

**Review by cdkini — APPROVED** [review]

**Comment by ghost** [comment]
<details open>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=7321&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

#### 

[Review these changes using an interactive CodeSee Map](https://app.codesee.io/r/reviews?pr=7321&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
CodeSee Map legend
</details>

**Comment by Shinnnyshinshin** [comment]
Thanks for your help with this @DrewHoo I did the linting and set auto-merge to True. PR should be going in in the next hour or so

**Comment by DrewHoo** [comment]
@Shinnnyshinshin you're too kind!

**Pull request merged**